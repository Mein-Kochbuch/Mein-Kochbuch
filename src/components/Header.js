import React from "react";
import {StyleSheet, Text, View} from "react-native";

export default function Header({title}) {

    return (
        <View style={styles.container}>
            <Text style={styles.heading}>
                {title}
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        marginVertical: 12,
        display: "flex",
        justifyContent: "center",
    },
    heading: {
        textAlign: "center",
        fontSize: 50,
    }
})
