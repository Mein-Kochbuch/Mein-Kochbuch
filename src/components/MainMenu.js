import React from "react";
import {StyleSheet, View} from "react-native";
import MenuButton from "./MenuButton";

export default function MainMenu() {

    return (
        <View style={styles.container}>
            <View style={styles.topmenu}>
                <MenuButton style={{margin: 12}}
                            key={"my-recipes-button"}
                            text={"My Recipes"}
                            path={"myrecipes"}
                />
                <MenuButton style={{margin: 12}}
                            key={"all-recipes-button"}
                            text={"All Recipes"}
                            path={"recipes"}
                />
                <MenuButton style={{margin: 12}}
                            key={"cookbooks-button"}
                            text={"Cookbooks"}
                            path={"cookbooks"}
                />
                <MenuButton style={{margin: 12}}
                            key={"ad-recipe-button"}
                            text={"Ad Recipe"}
                            path={"addrecipe"}
                />
            </View>
            <View style={styles.bottommenu}>
                <MenuButton key={"settings-button"} text={"Settings"}/>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        display: "flex",
        alignItems: "center",
        flexDirection: "column",
        height: "80%",
    },
    topmenu: {},
    bottommenu: {
        marginTop: "auto",
    }
})
