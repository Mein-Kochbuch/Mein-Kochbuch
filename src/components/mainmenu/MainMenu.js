import React from "react";
import {View} from "react-native";
import MenuButton from "./MenuButton";
import styled from "styled-components/native";

export default function MainMenu() {

    return (
        <StyledMenu>
            <View>
                <MenuButton
                    key={"my-recipes-button"}
                    text={"My Recipes"}
                    path={"myrecipes"}
                />
                <MenuButton
                    key={"all-recipes-button"}
                    text={"All Recipes"}
                    path={"recipes"}
                />
                <MenuButton
                    key={"cookbooks-button"}
                    text={"Cookbooks"}
                    path={"cookbooks"}
                />
                <MenuButton
                    key={"ad-recipe-button"}
                    text={"Ad Recipe"}
                    path={"addrecipe"}
                />
            </View>
            <StyledBottomMenu>
                <MenuButton key={"settings-button"} text={"Settings"} path={"settings"}/>
            </StyledBottomMenu>
        </StyledMenu>
    )
}

const StyledBottomMenu = styled.View`
  margin-top: auto;
`

const StyledMenu = styled.View`
  display: flex;
  align-items: center;
  flex-direction: column;
  height: 80%;
`
