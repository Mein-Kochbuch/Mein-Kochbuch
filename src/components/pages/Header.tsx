import React from "react";
import {StyleSheet, Text, View} from "react-native";

interface HeaderProps {
    title: string
}

export default function Header({title}: HeaderProps) {

    return (
        <View style={styles.container}>
            <Text style={styles.heading}>
                {title}
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        marginVertical: 12,
        display: "flex",
        justifyContent: "center",
    },
    heading: {
        textAlign: "center",
        fontSize: 50,
    }
})
