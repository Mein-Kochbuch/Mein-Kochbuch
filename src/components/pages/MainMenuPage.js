import React from "react";
import MainMenu from "../mainmenu/MainMenu";
import Header from "../list/Header";
import {StyleSheet, View} from "react-native";

export default function MainMenuPage() {


    return (
        <View style={styles.container}>
            <Header title={"My-Cookbook"}/>
            <MainMenu/>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        height: "100%",
    },
})
