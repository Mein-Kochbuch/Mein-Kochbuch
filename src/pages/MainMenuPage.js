import React from "react";
import MainMenu from "../components/mainmenu/MainMenu";
import Header from "../components/list/Header";
import {StyleSheet, View} from "react-native";

export default function MainMenuPage() {


    return (
        <View style={styles.container}>
            <Header title={"My-Cookbook"}/>
            <MainMenu/>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        height: "100%",
    },
})
