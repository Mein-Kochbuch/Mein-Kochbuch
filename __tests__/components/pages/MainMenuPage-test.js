import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Header from "../../../src/components/list/Header"
import MainMenuPage from "../../../src/components/pages/MainMenuPage";
import MainMenu from "../../../src/components/mainmenu/MainMenu";

jest.mock("../../../src/components/mainmenu/MainMenu", () => {
    return (props) => {
        return <div {...props}/>
    }
});

jest.mock("../../../src/components/list/Header", () => {
    return (props) => {
        return <div {...props}/>
    }
});


it('MainMenuPage Test', () => {

    const component = renderer.create(
        <MainMenuPage/>
    );

    expect(component.toJSON()).toMatchSnapshot()
    expect(component.toJSON().children.length).toBe(2)
    const testInstance = component.root
    expect(testInstance.findByType(Header).props.title).toBe("My-Cookbook")
    expect(testInstance.findByType(MainMenu)).not.toBeNull()
})
