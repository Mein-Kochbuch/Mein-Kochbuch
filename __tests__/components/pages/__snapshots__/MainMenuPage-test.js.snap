// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuPage Test 1`] = `
<View
  style={
    Object {
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
        "marginVertical": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 50,
          "textAlign": "center",
        }
      }
    >
      My-Cookbook
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "height": "80%",
        },
      ]
    }
  >
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 12,
            "marginLeft": 12,
            "marginRight": 12,
            "marginTop": 12,
            "minWidth": 300,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 30,
                "textAlign": "center",
              },
            ]
          }
        >
          My Recipes
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 12,
            "marginLeft": 12,
            "marginRight": 12,
            "marginTop": 12,
            "minWidth": 300,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 30,
                "textAlign": "center",
              },
            ]
          }
        >
          All Recipes
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 12,
            "marginLeft": 12,
            "marginRight": 12,
            "marginTop": 12,
            "minWidth": 300,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 30,
                "textAlign": "center",
              },
            ]
          }
        >
          Cookbooks
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 12,
            "marginLeft": 12,
            "marginRight": 12,
            "marginTop": 12,
            "minWidth": 300,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 30,
                "textAlign": "center",
              },
            ]
          }
        >
          Ad Recipe
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": "auto",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 12,
            "marginLeft": 12,
            "marginRight": 12,
            "marginTop": 12,
            "minWidth": 300,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 30,
                "textAlign": "center",
              },
            ]
          }
        >
          Settings
        </Text>
      </View>
    </View>
  </View>
</View>
`;
